### DISCUSSION TITLE
Overwrite current room temperature with other sensor

### DISCUSSION THREAD

**Discussion by forerunnert** [emsesp/EMS-ESP32#853] [Q&A] [answered]
Recently I decided to change my house primary heating from a Nefit Proline gas heater to a heatpump.
In doing so I removed the Nefit moduline 200 thermostat from the living room and changed it for a NSPanel smart device and use the heater for hot water and backup heating only.

This NSPanel has a  thermistor which I use to - through Home Assistant - feed the heatpump the current living room temperature and override its internal sensor.

Since the Moduline 200 currently hangs from a couple of wires besides the heater, rather than the living room, I would like to do the same for it.

So somehow sent the thermostat the living room temperature and have it use that rather than its internal sensor.

What would be a good way to do that?
Would I use the internal temperature offset (offset = external sensor - internal sensor)?

**Comment by caveman2003** [comment]
same here... have many temperature sensors (Homematic) and would like to use them ... any ideeas? 

Thanks :-)

**Reply by proddy** [reply]
I'm not 100% clear what you're trying to do, sorry.

**Comment by forerunnert** [comment]
I am trying to use an external temperature sensor to replace the internal thermostat's sensor.

I tried using the hc1_internal_temperature_offset to set the difference between the internal and external sensor.

That didn't work because the offset is applied to the internal sensor reading, so the next time an offset is in order it applies the offset to an already offset value so eventually that grows out of control.

I want to try using some node-red to keep track of the internal value without offsets, but that has the challenge to get the internal value without offset after restart.

It does sound something the EMS-ESP gateway could facilitate, what do you think?

**Reply by proddy** [reply]
could be, I'm not expert. Maybe @bbqkees or @MichaelDvP have an idea?

**Reply by MichaelDvP** [reply]
Sorry, i don't understand. There is no hc_1_internal_temperature_offset, there is only thermostat internal temperature offset. 
It's possible to calc a new offset with newoffset = ext_temp - roomtemp -oldoffset if it is not exceding the limit of +-5 degrees. But i think this is a eeprom saved value, writing every minute will kill the eeprom cell in ~ 2 years. But for this usage a dead eeprom cell do not matter.

**Comment by forerunnert** [comment, marked as answer]
Well I have made progress: I abandoned the idea of using the thermostat hardware and went for a software generic thermostat in Home Assistant.

I use the available room temperature sensor and have the software thermostat switch the ems-esp "heating activated" topic.

One quirk is that by using dhw hot water while "heating activated" the flow temperature resets to 5 degrees, so I added an automation to set the boiler heating flow temperature back to 50 degrees C every 15 seconds when "heating activated" (hot water flow temp set at 65 degrees C).

So "heating activated" => flow 50 => "hot water activated" => flow 65 => "hot water deactivated" => flow 5 => automation kicks in => flow 50

In my limited testing this setup allows for controlling room temperature within +/- 0.1 degrees C, but there isn't much heating demand at the moment here (hottest winter ever in Europe) so we'll see.
As I'm using this setup as backup heating only it may be good enough.

**Comment by JokerGermany** [comment]
Perhaps to clarify my use for this:
Our Thermostat is in the cellar at the boiler which is ... suboptimal...
Therefore another value e.g. from a TRV would be nice...
